projects/plugins/jetpack/modules/sharedaddy/sharing-sources.php

```diff
@@ -552,109 +602,95 @@ public function get_name() {
-		} elseif ( isset( $post_data['source_name'] ) ) {
-			$source_name = substr( $post_data['source_name'], 0, 200 );
+	/**
+	 * Build the HTML for the button.
+	 *
+	 * @param WP_Post $post The current post.
+	 * @return string The HTML for the button.
+	 */
+	public function get_display( $post ) {
+		$tracking_url = $this->get_process_request_url( $post->ID );
+		if ( false === stripos( $tracking_url, '?' ) ) {
+			$tracking_url .= '?';
 		} else {
-			$source_name = '';
+			$tracking_url .= '&';
 		}
+		$tracking_url .= 'share=email';
 
-		// Test email
-		$error = 1;	  // Failure in data
-		if ( empty( $post_data['source_f_name'] ) && $source_email && $target_email && $source_name ) {
-			/**
-			 * Allow plugins to stop the email sharing button from running the shared message through Akismet.
-			 *
-			 * @module sharedaddy
-			 *
-			 * @since 1.1.0
-			 *
-			 * @param bool true Should we check if the message isn't spam?
-			 * @param object $post Post information.
-			 * @param array $post_data Information about the shared message.
-			 */
-			if ( apply_filters( 'sharing_email_check', true, $post, $post_data ) ) {
-				$data = array(
-					'post'           => $post,
-					'source'         => $source_email,
-					'target'         => $target_email,
-					'name'           => $source_name,
-					'sharing_source' => $this,
-				);
-				// todo: implement an error message when email doesn't get sent.
-				/**
-				 * Filter whether an email can be sent from the Email sharing button.
-				 *
-				 * @module sharedaddy
-				 *
-				 * @since 1.1.0
-				 *
-				 * @param array $data Array of information about the shared message.
-				 */
-				if ( ( $data = apply_filters( 'sharing_email_can_send', $data ) ) !== false ) {
-					// Record stats
-					parent::process_request( $data['post'], $post_data );
-
-					/**
-					 * Fires when an email is sent via the Email sharing button.
-					 *
-					 * @module sharedaddy
-					 *
-					 * @since 1.1.0
-					 *
-					 * @param array $data Array of information about the shared message.
-					 */
-					do_action( 'sharing_email_send_post', $data );
-				}
-
-				// Return a positive regardless of whether the user is subscribed or not
-				if ( $ajax ) {
-?>
-<div class="response">
-	<div class="response-title"><?php _e( 'This post has been shared!', 'jetpack' ); ?></div>
-	<div class="response-sub"><?php printf( __( 'You have shared this post with %s', 'jetpack' ), esc_html( $target_email ) ); ?></div>
-	<div class="response-close"><a href="#" class="sharing_cancel"><?php _e( 'Close', 'jetpack' ); ?></a></div>
-</div>
-<?php
-				} else {
-					wp_safe_redirect( get_permalink( $post->ID ) . '?shared=email' );
-				}
+		$data_attributes = array(
+			'email-share-error-title' => __( 'Do you have email set up?', 'jetpack' ),
+			'email-share-error-text'  => __(
+				"If you're having problems sharing via email, you might not have email set up for your browser. You may need to create a new email yourself.",
```

Comment:
I wonder if we should link to a support doc of some kind, maybe one we'd set up on WordPress.com and Jetpack.com, to help people with this?

Reply:
> Where would the right location for that be?

I searched wordpress.com/support and jetpack.com/support/ and didn't see anything relevant, so I think we'll need new pages for this. I don't think this is necessarily a blocker.

It does look like we need to update some existing Jetpack documentation, specifically https://jetpack.com/blog/recaptcha-email-sharing-button/, which is no longer relevant, and removing the same content at the bottom of https://jetpack.com/support/sharing/.